cmd/pilot-agent/main.go

```diff
@@ -99,12 +83,46 @@ var (
 
 			glog.V(2).Infof("proxy role: %#v", role)
 
-			envoyProxy := envoy.MakeProxy(mesh, servicecluster, role.ServiceNode(), configpath, binarypath)
-			agent := proxy.NewAgent(envoyProxy, proxy.DefaultRetry)
-			watcher, err := envoy.NewWatcher(mesh, agent, role)
-			if err != nil {
+			proxyConfig := proxy.DefaultProxyConfig()
+			if config != "" {
+				if err := model.ApplyYAML(config, &proxyConfig); err != nil {
+					return multierror.Prefix(err, "failed to apply config YAML: ")
+				}
+			}
+
+			if proxyConfig.StatsdUdpAddress != "" {
+				if addr, err := proxy.ResolveAddr(proxyConfig.StatsdUdpAddress); err == nil {
+					proxyConfig.StatsdUdpAddress = addr
+				} else {
+					return err
+				}
+			}
+
+			if err := model.ValidateProxyConfig(&proxyConfig); err != nil {
 				return err
 			}
+
+			glog.V(2).Infof("effective config: %s", proxyConfig.String())
+
+			certs := []envoy.CertSource{
+				{
+					Directory: proxy.AuthCertsPath,
+					Files:     []string{proxy.CertChainFilename, proxy.KeyFilename, proxy.RootCertFilename},
+				},
+			}
+
+			if role.Type == proxy.Ingress {
+				certs = append(certs, envoy.CertSource{
+					Directory: proxy.IngressCertsPath,
+					Files:     []string{"tls.crt", "tls.key"},
```

Comment:
`tls.crt` and `tls.key` should also be defined in `proxy` package for consistency with the sidecar key/certs?

Reply:
Ok, will fix.

Reply:
Done